doc/index.rst

```diff
@@ -1,8 +1,45 @@
-Plutus
-======
+Plutus Programming and Reference Guide
+=============================================
+
+About Plutus
+--------------------
+
+Plutus consists of Plutus Core, the programming, scripting and assembly language; 
+and Plutus Tx, the compiler that compiles the Haskell source code into Plutus Core 
+to form the on-chain part of a contract application. All of this is used in combination 
+to write Plutus scripts that run on the Cardano blockchain. Compiled Plutus Core 
+scripts or programs are able to interact with the Cardano ledger through the 
+ledger interface. 
+
+The Plutus Core Repository
+----------------------------------
+
+The `Plutus Core Repository <https://github.com/input-output-hk/plutus>`_ contains 
+the implementation, specification, and mechanized metatheory of Plutus Core. 
+It also contains Plutus Tx. 
+
+About This Documentation
+---------------------------------
+
+The purpose of this documentation is to serve as an introduction to the Plutus language 
+and to programming in Plutus. It is meant to be educational, so it includes explanations, 
+tutorials and how-to instructions. It is also intended as a reference for beginning and 
+experienced developers. 
+
+Intended Audience
+---------------------------
+
+The intended audience of this documentation includes DApp developers, Plutus developers, 
+developers writing smart contracts, Haskell developers who want to write Plutus scripts, 
+and developers who want to learn to use the compiler, Plutus Tx. It is also designed for 
+developers who want to learn Plutus directly from the team that developed the Plutus language. 
+
+Because Plutus is built on top of Haskell, it is helpful to have a background in Haskell. 
+However, this document is also intended for developers who may not know Haskell very well. 
+
+This guide is also meant for people who need to reference and access specifications, 
```

Comment:
Is it? I guess we link to them.

Reply:
Actually it seems like links to specifications is very light. I'd like to consider more carefully what the appropriate scope for this documentation is on this point. Do you think we should make a more obvious effort to link to specifications and the like? We currently have a "Reference" section, and I think that section has some Reference sort of content, but not heavily so.

Reply:
Can we say how we will cater to the audience that doesn't know haskell well - e.g., by assuming a certain level and explaining advanced concepts or by linking to background material etc.

A key specification to link to is the ledger spec, as well as the plutus core spec.

Reply:
Honestly I can't pinpoint where/how I got the impression that we are targeting this documentation to people who may not have much understanding of Haskell already. I'd like to revisit this question. What is the minimum level of technical comprehension of Haskell that someone needs in order to get something helpful from it? What do we want to say about that, if anything?

Reply:
@jmchapman Thanks for mentioning the ledger spec and plutus core spec. I'll update with links to those.

Reply:
I don't think we should be targeting people who don't know Haskell. Some of the conceptual explanation stuff may be useful for them, though.

Reply:
I changed this to say that some knowledge of Haskell is expected.